### PR TITLE
feat(ecocredit/core): UpdateProjectAdmin/Metadata server impls

### PR DESCRIPTION
**Pull request by technicallyty**
## Description

- adds server impl for UpdateProjectAdmin and UpdateProjectMetadata

ref: #979 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/server/core/update_project_admin_test.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	s := setupBase(t)
+	projectId := "VERRA1"
+	assert.NilError(t, s.stateStore.ProjectInfoTable().Insert(s.ctx, &api.ProjectInfo{
+		Name:            projectId,
+		Admin:           s.addr,
+		ClassId:         1,
+		ProjectLocation: "US-NY",
+		Metadata:        "hi",
+	}))
+	newAdmin := sdk.AccAddress("addr1--------------------")
```
Comment (ryanchristo) [inline x/ecocredit/server/core/update_project_admin_test.go:25]:
```suggestion
	newAdmin := sdk.AccAddress("addr1")
```

File: x/ecocredit/server/core/update_project_admin_test.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	s := setupBase(t)
+	projectId := "VERRA1"
+	assert.NilError(t, s.stateStore.ProjectInfoTable().Insert(s.ctx, &api.ProjectInfo{
+		Name:            projectId,
+		Admin:           s.addr,
+		ClassId:         1,
+		ProjectLocation: "US-NY",
+		Metadata:        "hi",
+	}))
+	notAdmin := sdk.AccAddress("addr1--------------------")
```
Comment (ryanchristo) [inline x/ecocredit/server/core/update_project_admin_test.go:50]:
```suggestion
	notAdmin := sdk.AccAddress("addr1c")
```

File: x/ecocredit/server/core/update_project_metadata.go [hunk 3]
```diff
@@ -3,10 +3,29 @@ package core
...
 
+// UpdateProjectMetadata updates the project metadata.
 func (k Keeper) UpdateProjectMetadata(ctx context.Context, req *core.MsgUpdateProjectMetadata) (*core.MsgUpdateProjectMetadataResponse, error) {
-	//TODO implement me
-	panic("implement me")
+	admin, err := sdk.AccAddressFromBech32(req.Admin)
+	if err != nil {
+		return nil, err
+	}
+	project, err := k.stateStore.ProjectInfoTable().GetByName(ctx, req.ProjectId)
```
Comment (ryanchristo) [inline x/ecocredit/server/core/update_project_metadata.go:18]: Oof. This is a bit confusing. Should be sorted with #992.

File: x/ecocredit/server/core/update_project_metadata_test.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	s := setupBase(t)
+	projectId := "VERRA1"
+	assert.NilError(t, s.stateStore.ProjectInfoTable().Insert(s.ctx, &api.ProjectInfo{
+		Name:            projectId,
+		Admin:           s.addr,
+		ClassId:         1,
+		ProjectLocation: "US-NY",
+		Metadata:        "hi",
+	}))
+	notAdmin := sdk.AccAddress("addr1--------------------")
```
Comment (ryanchristo) [inline x/ecocredit/server/core/update_project_metadata_test.go:50]:
```suggestion
	notAdmin := sdk.AccAddress("addr1")
```

**Review by ryanchristo — APPROVED** [review]
Looks good to me. Just a few nits.

File: x/ecocredit/server/core/update_project_admin.go [hunk 5]
```diff
@@ -3,10 +3,35 @@ package core
...
+		return nil, sdkerrors.ErrInvalidRequest.Wrapf("could not get project with id %s: %s", req.ProjectId, err.Error())
+	}
+	if !sdk.AccAddress(project.Admin).Equals(admin) {
+		return nil, sdkerrors.ErrUnauthorized.Wrapf("%s is not the admin of project %s", req.Admin, req.ProjectId)
+	}
+	project.Admin = newAdmin
+	if err := k.stateStore.ProjectInfoTable().Update(ctx, project); err != nil {
+		return nil, err
+	}
+
```
Comment (aleem1314) [inline x/ecocredit/server/core/update_project_admin.go:35]: It looks like we don't have any events defined.

**Review by aleem1314 — APPROVED** [review]
lgtm

**Pull request merged**